src/overlays/actors/ovl_En_Torch2/z_en_torch2.c

```diff
@@ -27,21 +66,720 @@ const ActorInit En_Torch2_InitVars = {
+                                sStickTilt = 127.0f;
+                                sStickAngle += sStickAngle; // Not loaded from pointer
+                            } else if (player->swordAnimation <= LEFT_COMBO_2H) {
+                                sStickTilt = 127.0f;
+                                sStickAngle += -0x4000; // Not loaded from pointer
+                            } else if (player->swordAnimation <= HAMMER_SIDE) {
+                                input->cur.button = BTN_R;
+                            } else if (player->swordAnimation <= BIG_SPIN_2H) {
+                                EnTorch2_Backflip(this, input, &this->actor);
+                            } else {
+                                EnTorch2_Backflip(this, input, &this->actor);
+                            }
+                            if (CHECK_BTN_ANY(input->cur.button, BTN_A | BTN_R) && (this->swordState == 0) &&
+                                player->swordState) {
+                                sCounterState = 1;
+                            }
+                        }
+                    } else {
+                        /*
+                         * Handles movement and attacks when not reacting to Link's actions
+                         */
+                        sStickAngle = this->actor.yawTowardsLink;
+                        sp50 = 0.0f;
+                        if ((90.0f >= this->actor.xzDistFromLink) && (this->actor.xzDistFromLink > 70.0f) &&
+                            (ABS(sp5A) >= 0x7800) &&
+                            ((this->actor.unk_10C != 0) || !(player->stateFlags1 & 0x00400000))) {
+                            EnTorch2_SwingSword(globalCtx, input, this);
+                        } else if (((this->actor.xzDistFromLink <= 70.0f) ||
+                                    ((this->actor.xzDistFromLink <= 80.0f + sp50) && player->swordState)) &&
+                                   (this->swordState == 0)) {
+                            if (!EnTorch2_SwingSword(globalCtx, input, this) && (this->swordState == 0) &&
+                                (sCounterState == 0)) {
+                                EnTorch2_Backflip(this, input, &this->actor);
+                            }
+                        } else if (this->actor.xzDistFromLink <= (50 + sp50)) {
+                            sStickTilt = 127.0f;
+                            sStickAngle = this->actor.yawTowardsLink;
+                            if (this->actor.unk_10C == 0) {
+                                Math_SmoothScaleMaxMinS(&sStickAngle, player->actor.shape.rot.y + 0x7FFF, 1, 0x2328, 0);
+                            }
+                        } else if ((100.0f + sp50) < this->actor.xzDistFromLink) {
+                            if (!player->swordState || (player->swordAnimation < SPIN_ATTACK_1H) ||
+                                (player->swordAnimation > BIG_SPIN_2H) || (this->actor.xzDistFromLink >= 280.0f)) {
```

Comment:
swordState

Reply:
oh, lol, searched for this->swordState. Forgot there's also player->swordState.